client/src/components/ShoppingLists.js

```diff
@@ -24,37 +24,75 @@ const useStyles = makeStyles((theme) => ({
+        }
         window.scrollTo(0, 0);
     }, []);
 
     return(
         <section className={classes.root}>
-            <Container maxWidth="md" component="main" className={classes.TopContent}>
-                <Typography component="h1" variant="h4" align="center" color="textPrimary" className={classes.typoHeading} gutterBottom>
-                Add new item:
-                </Typography>
-                <DropDownListBar/>    
-            </Container>
+            {userLists ? (
+                null    
+            ) : (
+                <Container maxWidth="md" component="main" className={classes.TopContent}>
+                    <Typography component="h1" variant="h4" align="center" color="textPrimary" className={classes.typoHeading} gutterBottom>
+                    Add new item:
+                    </Typography>
+                    <DropDownListBar/>    
+                </Container>
+            )}
+            
             <Container maxWidth="md" component="main">
-                <Typography variant="h5" align="left" color="textPrimary" component="p"  style={{fontWeight: 'bold'}}>
-                My Shopping Lists:
-                </Typography>
+                {userLists ? (
+                    <Typography variant="h5" align="left" color="textPrimary" component="p"  style={{fontWeight: 'bold'}}>
+                    {props.userName}'s Shopping Lists:
+                    </Typography>
+                ) : (
+                    <Typography variant="h5" align="left" color="textPrimary" component="p"  style={{fontWeight: 'bold'}}>
+                    My Shopping Lists:
+                    </Typography>
+                )}
                 <br/>
-                <Grid container spacing={5}>
-                {shListsContext.lists.map((list) => (
-                    <Grid item key={list._id} xs={12} sm={6} md={4}>
-                        <ListCard list={list}/>
+                {userLists ? (
+                    <Grid container spacing={5}>
+                    {userLists.map((list) => (
+                        <Grid item key={list._id} xs={12} sm={6} md={4}>
+                            <ListCard list={list} userName={props.userName} hidden={true}/>
+                        </Grid>
+                    ))}
                     </Grid>
-                ))}
-                    <Grid item xs={12} sm={6} md={4}>
-                        <AddNewListCard/>
+
+                ) : (
+
+                    <Grid container spacing={5}>
+                    {shListsContext.lists.map((list) => (
+                        <Grid item key={list._id} xs={12} sm={6} md={4}>
+                            <ListCard list={list}/>
+                        </Grid>
+                    ))}
+                        <Grid item xs={12} sm={6} md={4}>
+                            <AddNewListCard/>
+                        </Grid>
                     </Grid>
-                </Grid>
+                )}
             </Container>
             
```

Comment:
May I know the reason why you decided to reuse the entire ShoppingLists.js but not to create a new container component and only reuse the functional components like ListCard.js and ProductCard.js? Your design seems to be opposite to what Eric said about modularity on Slack.

Reply:
Haha yeah! I understand. Although, it would be a lot of code duplication. But are we implementing modularity in this project? If yes, then we can try a bit of code refactoring and abstraction but without any errors and failures. (or else we would not have anything working to show on Friday's Final Presentation lol)

Reply:
I understand your concerns, and I think this branch is already good enough to merge. However, regardless of merging branch and presentation, there is still one thing I want to bring up and discuss. That is, the code snippets you added into this file are almost enough to build a whole new container-component. you only saved library imports, useStyles, the first <section>, the top most <container> line, but you need to increase more conditional sentences. Then, if the public page has different fetching, rendering or context-managing logic from the private page, you may require more flags or conditional sentences.

Reply:
Whatever, for now, we just keep it and go for other tasks.